src/cortex_cli/cortex_cli.py

```diff
@@ -650,6 +659,73 @@ def _validate_cortex_cli_auth(no_auth, config_file) -> Optional[str]:
     return str(tokens_file)
 
 
+def _validate_measurements(input_circuit, results) -> Dict[str, List[str]]:
+    """Checks if the results of a circuit execution match the expected measurements in the input circuit."""
+    if results.measurements is None:
+        raise click.ClickException(f'No measurements obtained from backend. Job status is ${results.status}')
+
+    measured_qubits = {}
+    for instruction in input_circuit.instructions:
+        if instruction.name == 'measurement':
+            measured_qubits[instruction.args['key']] = list(instruction.qubits)
+
+    for measurement in results.measurements:
+        for measurement_name in measurement:
+            if measurement_name not in measured_qubits:
+                raise click.ClickException(
+                    'Measurements obtained from the backend \
+                     do not match measurements in the circuit.'
+                )
+
+    return measured_qubits
+
+
+def _human_readable_frequencies_output(input_circuit_name, measured_qubits, shots, results) -> str:
```

Comment:
For frequencies assume you have a measurement key `k_1` that measures multiple qubits `QB1, QB2, QB3` than frequencies should look like
000: 0.25 
100: 0.25
010: 0.25
001: 0.25
...
otherwise we are missing important information.

Reply:
Also maybe it makes sense to sort it in decreasing order 🤔

Reply:
Updated